src/jobservice/period/enqueuer_test.go

```diff
@@ -120,16 +120,26 @@ func (suite *EnqueuerTestSuite) TestEnqueuer() {
 }
 
 func (suite *EnqueuerTestSuite) prepare() {
+	var cronSpec string
+
 	now := time.Now()
 	minute := now.Minute()
 
-	coreSpec := fmt.Sprintf("0-59 %d-%d * * * *", minute, minute+2)
+	// Make sure we at least schedule jobs in the 3 minutes cycle
+	switch minute {
+	case 58:
+		cronSpec = "0-59 58,59,0 * * * *"
+	case 59:
+		cronSpec = "0-59 59,0,1 * * * *"
+	default:
+		cronSpec = fmt.Sprintf("0-59 %d-%d * * * *", minute, minute+2)
+	}
```

Comment:
I think we should avoid such random test, it will make the code coverage unstable.
It's better to use a for loop to verify each fake time minutes

Reply:
Random testing is more realistic.

Reply:
> om test, it will make the code coverage unstable.
> It's better to use a for loop to verify each fake time minutes

The case here is to make sure there are jobs enqueued based on the current minute. I don't think the `for` loop is suitable for this case.